### DISCUSSION TITLE
Value cannot be null?

### DISCUSSION THREAD

**Discussion by naviprobeha** [microsoft/AL-Go#1276] [Q&A]
Hello!

I have had an issue publishing a certain 3rd party app in my Al-go repos, the last 3 weeks. The error is "Value cannot be null". Does somebody have a clue?

Stacktrace:

Publishing C:\ProgramData\BcContainerHelper\Extensions\bc11497069871\c3a7b04e-1f5c-404e-a78f-49c062b2b932\K3 Pebblestone_24.3.0.0_runtime_debug.app
  Value cannot be null. (Parameter 'input')
  
  Exception Script Stack Trace:
  at <ScriptBlock>, <No file>: line 41
  
  PowerShell Call Stack:
  at Invoke-ScriptInBcContainer, C:\ProgramData\BcContainerHelper\6.0.26\BcContainerHelper\ContainerHandling\Invoke-ScriptInNavContainer.ps1: line 71
  at <ScriptBlock>, C:\ProgramData\BcContainerHelper\6.0.26\BcContainerHelper\AppHandling\Publish-NavContainerApp.ps1: line 379

**Comment by jonaswre** [comment]
I think I had a similar error. Could you provide more of the log? I think the issue was that the partner changed the URL where we downloaded the app from.

**Comment by naviprobeha** [comment]
Here is the complete "Install apps" log section. All of the apps are zipped in pebblestone24.zip, and some are working and some are not.
Right above of the error message is an example of another app that is working.

Downloading C:\Users\github_service\AppData\Local\Temp\499e3ad0-9c3e-4714-baf9-9d515439fa13\pebblestone24.zip
  Downloading C:\Users\github_service\AppData\Local\Temp\ef29d054-647f-4686-b7b4-cdd3e7d0d9e2\swebase.zip
  SmartApps_LicenseProvider_22.2.113403.0.app (AppId=263e0154-82ae-44b6-82c1-21fe4201f54a) is skipped as it is not referenced
  Programekonomi Svenska AB_SweBase_22.5.79.0.app (AppId=b830ef47-c998-4e30-bb03-52be437f36f6) is skipped as it is not referenced
  Copy C:\ProgramData\BcContainerHelper\Extensions\bc11497069871\c3a7b04e-1f5c-404e-a78f-49c062b2b932\K3 Barcodes_7.3.0.0_runtime_debug.app to C:\actions-runner\_work\navipro\navipro\.buildartifacts\Dependencies
  Publishing C:\ProgramData\BcContainerHelper\Extensions\bc11497069871\c3a7b04e-1f5c-404e-a78f-49c062b2b932\K3 Barcodes_7.3.0.0_runtime_debug.app
  Synchronizing K3 Barcodes on tenant default
  Installing K3 Barcodes on tenant default
  App K3 Barcodes_7.3.0.0_runtime_debug.app successfully published
  Copy C:\ProgramData\BcContainerHelper\Extensions\bc11497069871\c3a7b04e-1f5c-404e-a78f-49c062b2b932\K3 Buying Groups_7.3.0.0_runtime_debug.app to C:\actions-runner\_work\navipro\navipro\.buildartifacts\Dependencies
  Publishing C:\ProgramData\BcContainerHelper\Extensions\bc11497069871\c3a7b04e-1f5c-404e-a78f-49c062b2b932\K3 Buying Groups_7.3.0.0_runtime_debug.app
  Synchronizing K3 Buying Groups on tenant default
  Installing K3 Buying Groups on tenant default
  App K3 Buying Groups_7.3.0.0_runtime_debug.app successfully published
  Copy C:\ProgramData\BcContainerHelper\Extensions\bc11497069871\c3a7b04e-1f5c-404e-a78f-49c062b2b932\K3 Core Application_7.3.0.0_runtime_debug.app to C:\actions-runner\_work\navipro\navipro\.buildartifacts\Dependencies
  Publishing C:\ProgramData\BcContainerHelper\Extensions\bc11497069871\c3a7b04e-1f5c-404e-a78f-49c062b2b932\K3 Core Application_7.3.0.0_runtime_debug.app
  Synchronizing K3 Core Application on tenant default
  Installing K3 Core Application on tenant default
  App K3 Core Application_7.3.0.0_runtime_debug.app successfully published
  Copy C:\ProgramData\BcContainerHelper\Extensions\bc11497069871\c3a7b04e-1f5c-404e-a78f-49c062b2b932\K3 Pebblestone Payments_24.3.0.0_runtime_debug.app to C:\actions-runner\_work\navipro\navipro\.buildartifacts\Dependencies
  Publishing C:\ProgramData\BcContainerHelper\Extensions\bc11497069871\c3a7b04e-1f5c-404e-a78f-49c062b2b932\K3 Pebblestone Payments_24.3.0.0_runtime_debug.app
  Synchronizing K3 Pebblestone Payments on tenant default
  Installing K3 Pebblestone Payments on tenant default
  App K3 Pebblestone Payments_24.3.0.0_runtime_debug.app successfully published
  Copy C:\ProgramData\BcContainerHelper\Extensions\bc11497069871\c3a7b04e-1f5c-404e-a78f-49c062b2b932\K3 Pebblestone Delivery Plan_24.3.0.0_runtime_debug.app to C:\actions-runner\_work\navipro\navipro\.buildartifacts\Dependencies
  Publishing C:\ProgramData\BcContainerHelper\Extensions\bc11497069871\c3a7b04e-1f5c-404e-a78f-49c062b2b932\K3 Pebblestone Delivery Plan_24.3.0.0_runtime_debug.app
  Synchronizing K3 Pebblestone Delivery Plan on tenant default
  Installing K3 Pebblestone Delivery Plan on tenant default
  App K3 Pebblestone Delivery Plan_24.3.0.0_runtime_debug.app successfully published
  Copy C:\ProgramData\BcContainerHelper\Extensions\bc11497069871\c3a7b04e-1f5c-404e-a78f-49c062b2b932\K3 Pebblestone_24.3.0.0_runtime_debug.app to C:\actions-runner\_work\navipro\navipro\.buildartifacts\Dependencies
  Publishing C:\ProgramData\BcContainerHelper\Extensions\bc11497069871\c3a7b04e-1f5c-404e-a78f-49c062b2b932\K3 Pebblestone_24.3.0.0_runtime_debug.app
  Value cannot be null. (Parameter 'input')
  
  Exception Script Stack Trace:
  at <ScriptBlock>, <No file>: line 41
  
  PowerShell Call Stack:
  at Invoke-ScriptInBcContainer, C:\ProgramData\BcContainerHelper\6.0.26\BcContainerHelper\ContainerHandling\Invoke-ScriptInNavContainer.ps1: line 71
  at <ScriptBlock>, C:\ProgramData\BcContainerHelper\6.0.26\BcContainerHelper\AppHandling\Publish-NavContainerApp.ps1: line 379
  at Publish-BcContainerApp, C:\ProgramData\BcContainerHelper\6.0.26\BcContainerHelper\AppHandling\Publish-NavContainerApp.ps1: line 154
  at <ScriptBlock>, C:\ProgramData\BcContainerHelper\6.0.26\BcContainerHelper\AppHandling\Run-AlPipeline.ps1: line 873
  at <ScriptBlock>, C:\ProgramData\BcContainerHelper\6.0.26\BcContainerHelper\AppHandling\Run-AlPipeline.ps1: line 1304
  at <ScriptBlock>, C:\ProgramData\BcContainerHelper\6.0.26\BcContainerHelper\AppHandling\Run-AlPipeline.ps1: line 1234
  at <ScriptBlock>, C:\ProgramData\BcContainerHelper\6.0.26\BcContainerHelper\AppHandling\Run-AlPipeline.ps1: line 977
  at Run-AlPipeline, C:\ProgramData\BcContainerHelper\6.0.26\BcContainerHelper\AppHandling\Run-AlPipeline.ps1: line 939
  at <ScriptBlock>, C:\actions-runner\_work\_actions\microsoft\AL-Go-Actions\v4.1\RunPipeline\RunPipeline.ps1: line 368
  at <ScriptBlock>, C:\actions-runner\_work\_temp\3a4a20ba-a4e0-45b4-8a78-ff0f569d98a1.ps1: line 4
  at <ScriptBlock>, <No file>: line 1
  
  Container Free Physical Memory: 8.5Gb
  Disk C: Free 124Gb from 127Gb

**Comment by freddydk** [comment]
Is it always the same app failing: `K3 Pebblestone_24.3.0.0_runtime_debug.app` ?

**Comment by naviprobeha** [comment]
Yes, correct. In multiple repos.


/Håkan
NaviPro AB

25 okt. 2024 kl. 05:10 skrev Freddy Kristiansen ***@***.***>:

﻿

Is it always the same app failing: K3 Pebblestone_24.3.0.0_runtime_debug.app ?

—
Reply to this email directly, view it on GitHub<https://github.com/microsoft/AL-Go/discussions/1276#discussioncomment-11047444>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AM6P4XLVNEJMAAPXXUY36PTZ5GZB7AVCNFSM6AAAAABQRABT72VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMBUG42DINA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

Med vänlig hälsning/Best regards,
Håkan Bengtsson
Systemarkitekt

[https://navipro.se/wp-content/uploads/2024/05/NaviPro_pbEvidi-logo500px.jpg]<https://navipro.se/pressrelease-navipro-blir-en-del-av-evidi/>    E: ***@***.******@***.***>
M: +46730698914
P: +4633208914
A: NaviPro AB, Lilla Brogatan 35, 50335 Borås, SE
www.navipro.se<http://www.navipro.se>
[https://www.mail-signatures.com/signature-generator/img/templates/christmas-2022-1/fb.png]<https://www.facebook.com/NaviProAB/>        [https://www.mail-signatures.com/signature-generator/img/templates/christmas-2022-1/ln.png] <https://www.linkedin.com/company/navipro-ab>       [https://www.mail-signatures.com/signature-generator/img/templates/christmas-2022-1/it.png] <https://www.instagram.com/naviproab>
[https://navipro.se/wp-content/uploads/2024/06/PoweredByEvidi-banner.gif]<https://navipro.se/>

Innehållet i detta e-postmeddelande är endast avsett för den mottagare som anges i meddelandet. Det är inte tillåtet att dela någon del av detta meddelande med tredje part utan skriftligt medgivande från avsändaren. Om du fick det här meddelandet av misstag, vänligen svara på det här meddelandet och ta bort det, så att vi kan säkerställa att ett sådant misstag inte inträffar i framtiden.

**Comment by freddydk** [comment]
Would it be possible to get the .zip file with all these runtime packages?
Maybe you can ask K3 if you can share the .zip file with me (not publicly here) - my assumption is that something is wrong in that.

**Comment by naviprobeha** [comment]
Had to ask K3 for permission.

However: It turned out that the Publish-NAVApp in BC returned the exact same message, This message is apparently shown when trying to publish a runtime package built for an older version of BC. 24.4 => 25.0. This hovever does not explain why the other apps worked fine.

This means for us that we will be getting this error during a window of 1,5 months due to the fact that K3 is releasing afterwards.
